app/jobs/get_next_ai_message_job.rb

```diff
@@ -189,6 +192,13 @@ def call_tools_before_wrapping_up
         index: index += 1,
         processed_at: Time.current,
       )
+
+      content_hash = JSON.parse(tool_message[:content])
+
+      if content_hash.has_key?('url_of_dalle_generated_image')
+        url_of_dalle_generated_image = content_hash['url_of_dalle_generated_image']
+      end
```

Comment:
The failing test is because content_hash is not always a hash. We need to check for that. In addition, I think we should assume that tool calling could occasionally return invalid JSON.

```suggestion
      parsed = JSON.parse(tool_message[:content]) rescue nil

      if parsed.is_a?(Hash) && parsed.has_key?("url_of_dalle_generated_image")
        url_of_dalle_generated_image = parsed["url_of_dalle_generated_image"]
      end
```